lib/streamlit/runtime/scriptrunner/script_run_context.py

```diff
@@ -77,6 +78,9 @@ class ScriptRunContext:
     cursors: Dict[int, "streamlit.cursor.RunningCursor"] = field(default_factory=dict)
     script_requests: Optional[ScriptRequests] = None
 
+    _experimental_query_params_used = False
+    _production_query_params_used = False
```

Comment:
Looking at this now, I feel like the current way this is designed feels kinda janky.

We currently have to:
1. Have the code that's about to use the query params API set an internal variable of `ScriptRunContext`
2. Manually call `ctx.ensure_single_query_api_used()`

The two always happen at the same time, so why not combine both steps into (two) methods (`mark_experimental_query_params_used` or something like that) to do so that we don't have to muck around with an object's internal attributes + call a method too?